app/src/main/java/edu/agh/susgame/front/model/graph/GameGraph.kt

```diff
@@ -5,31 +5,45 @@ import edu.agh.susgame.front.model.PlayerId
 import edu.agh.susgame.front.util.Coordinates
 
 class GameGraph(
-    val nodes: MutableMap<NodeId, Node>,
-    val edges: MutableMap<EdgeId, Edge>,
+    val nodes: Map<NodeId, Node>,
+    val edges: Map<EdgeId, Edge>,
+    val players: Map<PlayerId, Player>,
     val paths: MutableMap<PlayerId, Path>,
-    val players: MutableMap<PlayerId, Player>,
-    val nodesToEdges: MutableMap<Pair<NodeId, NodeId>, EdgeId>,
-    var serverId: NodeId = NodeId(0),
+    val serverId: NodeId,
     val mapSize: Coordinates,
 ) {
-    fun addNode(node: Node) {
-        nodes[node.id] = node
+    companion object {
+        fun fromLists(
+            nodes: List<Node>,
+            edges: List<Edge>,
+            players: List<Player>,
+            serverId: NodeId,
+            mapSize: Coordinates,
+        ): GameGraph =
+            GameGraph(
+                nodes = nodes.associateBy { it.id },
+                edges = edges.associateBy { it.id },
+                players = players.associateBy { it.id!! },
```

Comment:
Używanie tego `!!` jest w ogólności niewskazane, bo jeśli będzie tam null, to zostanie rzucone `NullPointerException` i zarówno wszystko się wysypie, jak i komunikat o błędzie nie będzie mówił za dużo

Normalnie fajnie by to było jakoś otypować, żeby wymusić ustawione idki jako argument tej metody. Tutaj, jako że to jest używane tylko w mockowej implementacji, to imo wystarczy zupełnie coś w stylu:
```
players = players.associateBy {
    it.id ?: throw IllegalArgumentException("One of players had unset id")
},
```